api_tests/src/test_environment.ts

```diff
@@ -0,0 +1,130 @@
+        this.testRoot = path.join(this.projectRoot, "/api_tests");
+
+        // setup global variables
+        this.global.projectRoot = this.projectRoot;
+        this.global.testRoot = this.testRoot;
+        this.global.ledgerConfigs = {};
+        this.global.verbose =
+            this.global.process.argv.find(item => item.includes("verbose")) !==
+            undefined;
+
+        this.global.parityAccountMutex = new Mutex();
+
+        if (this.global.verbose) {
+            console.log(`Starting up test environment`);
+        }
+
+        const { ledgers, logDir } = this.extractDocblockParams(
+            this.docblockPragmas
+        );
+
+        this.logDir = path.join(this.projectRoot, "api_tests", "log", logDir);
+        await E2ETestEnvironment.cleanLogDir(this.logDir);
+
+        if (ledgers.length > 0) {
+            // setup ledgers
+            this.ledgerRunner = new LedgerRunner(
+                this.projectRoot,
+                this.logDir,
+                this.global
+            );
+
+            if (this.global.verbose) {
+                console.log(`Initializing ledgers : ${ledgers}`);
+            }
+            const ledgerConfig = await this.ledgerRunner.ensureLedgersRunning(
+                ledgers
+            );
+            this.global.tokenContract = ledgerConfig.ethereum.tokenContract;
+            this.global.ledgerConfigs = {
+                bitcoin: ledgerConfig.bitcoin,
+                ethereum: ledgerConfig.ethereum,
+            };
+        }
+        this.global.logRoot = this.logDir;
+    }
+
+    private static async cleanLogDir(logDir: string) {
+        await rimrafAsync(logDir);
+        await mkdirAsync(logDir, { recursive: true });
+    }
+
+    async teardown() {
+        await super.teardown();
+        if (this.global.verbose) {
+            console.log(`Tearing down test environment.`);
+        }
+        await this.cleanupAll();
+        if (this.global.verbose) {
+            console.log(`All teared down.`);
+        }
+    }
+
+    async cleanupAll() {
+        try {
+            if (this.ledgerRunner) {
+                await this.ledgerRunner.stopLedgers();
+            }
+        } catch (e) {
+            console.error("Failed to clean up resources", e);
+        }
+    }
+
+    // ignoring warnings within this functions as TS does not know that docblokPragmas may has these fields
+    private extractDocblockParams(
+        docblockPragmas: Record<string, string>
+    ): { logDir: string; ledgers: string[] } {
+        // @ts-ignore
+        const docblockLedgers = docblockPragmas.ledgers;
+        const ledgers = docblockLedgers ? docblockLedgers.split(",") : [];
+
+        // @ts-ignore
+        const logDir = this.docblockPragmas.logDir;
+        if (!logDir) {
+            throw new Error(
+                "Test file did not specify a log directory. Did you miss adding @logDir"
+            );
+        }
+
+        // @ts-ignore
+        return { ledgers, logDir };
```

Comment:
```suggestion
        return { ledgers, logDir };
```
I don't see why this one should be needed.